### PR TITLE
swarm/pss: fix data race in notify_test.go (TestStart)

### LINKED ISSUE #1158: Data race in pss/notify tests
# How to reproduce?

`go test -race -count 1 -run TestStart github.com/ethereum/go-ethereum/swarm/pss/notify`

# Stack trace

```
==================
WARNING: DATA RACE
Write at 0x00c00018dd40 by goroutine 16:
  github.com/ethereum/go-ethereum/swarm/pss/notify.TestStart()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/notify/notify_test.go:183 +0x1e84
  testing.tRunner()
      /usr/lib/go/src/testing/testing.go:827 +0x162

Previous read at 0x00c00018dd40 by goroutine 90:
  github.com/ethereum/go-ethereum/swarm/pss/notify.TestStart.func1()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/notify/notify_test.go:149 +0x303
  github.com/ethereum/go-ethereum/swarm/pss/notify.(*Controller).handleNotifyWithKeyMsg()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/notify/notify.go:340 +0x54d
  github.com/ethereum/go-ethereum/swarm/pss/notify.(*Controller).Handler()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/notify/notify.go:386 +0x22c
  github.com/ethereum/go-ethereum/swarm/pss/notify.(*Controller).Handler-fm()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/notify/notify.go:116 +0x99
  github.com/ethereum/go-ethereum/swarm/pss.(*Pss).executeHandlers()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/pss.go:492 +0x317
  github.com/ethereum/go-ethereum/swarm/pss.(*Pss).process()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/pss.go:465 +0x27b
  github.com/ethereum/go-ethereum/swarm/pss.(*Pss).handlePssMsg()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/pss.go:417 +0x704
  github.com/ethereum/go-ethereum/swarm/pss.(*Pss).handlePssMsg-fm()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/pss.go:274 +0x6d
  github.com/ethereum/go-ethereum/p2p/protocols.(*Peer).handleIncoming()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/p2p/protocols/protocol.go:374 +0xa8a
  github.com/ethereum/go-ethereum/p2p/protocols.(*Peer).Run()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/p2p/protocols/protocol.go:232 +0x46
  github.com/ethereum/go-ethereum/swarm/pss.(*Pss).Run()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/pss.go:274 +0x24f
  github.com/ethereum/go-ethereum/swarm/pss.(*Pss).Run-fm()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/swarm/pss/pss.go:264 +0x63
  github.com/ethereum/go-ethereum/p2p.(*Peer).startProtocols.func1()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/p2p/peer.go:361 +0x8d

Goroutine 16 (running) created at:
  testing.(*T).Run()
      /usr/lib/go/src/testing/testing.go:878 +0x659
  testing.runTests.func1()
      /usr/lib/go/src/testing/testing.go:1119 +0xa8
  testing.tRunner()
      /usr/lib/go/src/testing/testing.go:827 +0x162
  testing.runTests()
      /usr/lib/go/src/testing/testing.go:1117 +0x4ee
  testing.(*M).Run()
      /usr/lib/go/src/testing/testing.go:1034 +0x2ee
  main.main()
      _testmain.go:42 +0x221

Goroutine 90 (running) created at:
  github.com/ethereum/go-ethereum/p2p.(*Peer).startProtocols()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/p2p/peer.go:360 +0x2a8
  github.com/ethereum/go-ethereum/p2p.(*Peer).run()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/p2p/peer.go:210 +0x16a
  github.com/ethereum/go-ethereum/p2p.(*Server).runPeer()
      /home/frncmx/code/src/github.com/ethereum/go-ethereum/p2p/server.go:1002 +0x25d
==================
--- FAIL: TestStart (1.07s)
    testing.go:771: race detected during execution of test
FAIL
FAIL	github.com/ethereum/go-ethereum/swarm/pss/notify	1.112s
```

### PR DESCRIPTION
**Pull request by frncmx**
TestStart failed when all tests were run with race detection.
Simplest way to reproduce:
`go test -race -count 1 -run TestStart ./swarm/pss/notify`

The handler passed to `ctrlClient.Subscribe` is run in a goroutine,
started in `p2p.(*Peer).startProtocols()`. There were 2 problems/races
cased by the handler closure:
1. `t.Fatalf` was called from a goroutine
2. `updateMsg` passed to the handler (reda) was later modified in the
main thread (write)

fixes ethersphere/go-ethereum#1158

### CODE REVIEW

File: swarm/pss/notify/notify_test.go [hunk 1]
```diff
@@ -145,17 +145,24 @@ func TestStart(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+
+	copyOfUpdateMsg := make([]byte, len(updateMsg))
```
Comment (frncmx) [inline swarm/pss/notify/notify_test.go:149]: This copy is future proof, but I was thinking about just inlining `byte[]{}` in the lambda. What do you think?

Comment (janos) [reply swarm/pss/notify/notify_test.go:149]: I think that copy is ok, maybe a comment why do we need it, but not required.

**Review by janos — APPROVED** [review]
LGTM

**Review by nolash — APPROVED** [review]

**Comment by frncmx** [comment]
submitted upstream
https://github.com/ethereum/go-ethereum/pull/18518

**Pull request closed without merging**